File: backend/planta-smart-4.0/src/main/java/com/smart/appsa/model/enums/AndarBloco.java
package com.smart.appsa.model.enums;

import com.fasterxml.jackson.annotation.JsonCreator;

import jakarta.persistence.EnumeratedValue;

public enum AndarBloco {
    PRIMEIRO(1),
    SEGUNDO(2),
    TERCEIRO(3);

    @EnumeratedValue
    int value;

    @JsonCreator
    public static AndarBloco fromValue(String value) {
        Integer parsed = Integer.valueOf(value);
        for (AndarBloco p : values()) {
            if (parsed.equals(p.value)) return p;
        }
        throw new IllegalArgumentException("AndarBloco inválido: " + value);
    }

    private AndarBloco(Integer value) {
        this.value = value;
    }

    public Integer getValue() { return value; }
}

File: frontend/planta-smart-4.0/src/domain/enums/AndarBloco.ts
export enum AndarBloco {
  Primeiro,
  Segundo,
  Terceiro,
}

export const AndarBlocoToInt: Record<AndarBloco, number> = {
  [AndarBloco.Primeiro]: 1,
  [AndarBloco.Segundo]: 2,
  [AndarBloco.Terceiro]: 3,
};

export const AndarBlocoStringToEnum: Record<string, AndarBloco> = {
  PRIMEIRO: AndarBloco.Primeiro,
  SEGUNDO: AndarBloco.Segundo,
  TERCEIRO: AndarBloco.Terceiro,
};